app/models/work_activity_notification.rb

```diff
@@ -31,6 +38,17 @@ def wait_time
       end
     end
 
+    def check_from_state # check the previous state of the work
+      case work_activity.message
+      when /has been created/
+        :none
+      when /for revision/
+        :awaiting_approval
+      when /ready for review/
+        :draft
+      end
+    end
+
```

Comment:
This logic works but it's a bit brittle, I would suggest grabbing the "from" state from the database, for example with code like this:

```suggestion
    def check_from_state(work) # check the previous state of the work
      work_states = UserWork.where(work_id: work.id).sort_by(&:created_at).reverse
      if work_states.count > 1
        # states[0] is the current state, [1] is the previous state
        work_states[1].state.to_sym
      else
        :none
      end
    end
```